aten/src/ATen/native/Normalization.cpp

```diff
@@ -463,6 +463,11 @@ std::tuple<Tensor, Tensor, Tensor, Tensor, int64_t> _batch_norm_impl_index(
              std::make_tuple(2));
   }
 
+  if (input.is_mkldnn()) {
+    TORCH_CHECK(at::globalContext().userEnabledMkldnn(),
+        "MKLDNN batch_norm module can't run when mkldnn is disabled");
+  }
+
```

Comment:
For BatchNorm,
```
x = torch.randn(1, 3, 56, 56, dtype=torch.float)
bn = torch.nn.BatchNorm2d(3).float().train(False)
mkldnn_bn = mkldnn_utils.to_mkldnn(copy.deepcopy(bn))
with torch.backends.mkldnn.flags(enabled=True):
    y_mkldnn = bn(x)
```
1. torch.backends.mkldnn.flags(enabled=True), input and weight are MKLDNN tensor, it can works.
1. torch.backends.mkldnn.flags(enabled=True), input is dense tensor,  weight are MKLDNN tensor, it report a error:**mkldnn_tensor.is_mkldnn() INTERNAL ASSERT FAILED at "../aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp":56, please report a bug to PyTorch. mkldnn_to_dense expects MKL-DNN tensor input**.
3. torch.backends.mkldnn.flags(enabled=False), input and weight are MKLDNN tensor, it will report a error:**MKLDNN batch_norm module can't run when mkldnn is disabled**.
4. torch.backends.mkldnn.flags(enabled=False), input is dense tensor, weight are MKLDNN tensor, it will report a error:**mkldnn_tensor.is_mkldnn() INTERNAL ASSERT FAILED at "../aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp":56, please report a bug to PyTorch. mkldnn_to_dense expects MKL-DNN tensor input**.

Reply:
There should not be INTERNAL_ASSERTs here, right? Is it a real pytorch bug that you guys have to fix, or should the error message be different. Couple additional questions
1) why is mkldnn_to_dense attempted here, is it when converting output of the operation to dense tensor?
2) Why is behavior here different than in linear, where case 2 works, and case 4 errors out as expected, with expected error message?

Reply:
For the first one, it call in https://github.com/pytorch/pytorch/blob/7b0f867c4853af69b14bb516ba2a9dbff8f56d51/aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp#L54, which the error message is wrong, it shold be "itensor_from_mkldnn expects MKL-DNN tensor input"

For the second: the case 2 can be works for linear is that we always convert dense tensor to MKLDNN tensor for input, see https://github.com/pytorch/pytorch/blob/7b0f867c4853af69b14bb516ba2a9dbff8f56d51/torch/utils/mkldnn.py#L30. the case 4's error message need make some changes to be unified.